vm/src/stdlib/os.rs

```diff
@@ -862,6 +863,22 @@ pub fn os_openpty(vm: &VirtualMachine) -> PyResult {
     }
 }
 
+#[cfg(unix)]
+pub fn os_ttyname(fd: PyIntRef, vm: &VirtualMachine) -> PyResult {
+    use libc::ttyname;
+    if let Some(fd) = fd.as_bigint().to_i32() {
+        let name = unsafe { ttyname(fd) };
+        if !name.is_null() {
```

Comment:
Note: you might want to swap if-arms if the condition contains a `!` for readability:

```rust
        if name.is_null() {
            Err(vm.new_os_error(Error::last_os_error().to_string()))
        } else {
            let name = unsafe { CStr::from_ptr(name) }.to_str().unwrap();
            Ok(vm.ctx.new_str(name.to_owned()))
        }
```

This is a slight improvement, but not required. I think this change is good now.